File: Projekt1_cpp/Projekt1_cpp/WilczeJagody.cpp
#include "WilczeJagody.h"

WilczeJagody::WilczeJagody(pair<int, int>pole, Swiat* swiat)
	:Roslina(99, 0, pole, swiat) {}

Organizm* WilczeJagody::potomek(pair<int, int>pole) {
	return new WilczeJagody(pole, swiat);
}

void WilczeJagody::rysowanie() {
	cout << 'j';
}

string WilczeJagody::rodzajKolizji(Organizm* organizm) {
	if (organizm->getSila() >= getSila()) {
		cout << getNazwa() << " ( " << getPole().first + 1 << "," << getPole().second + 1 << " ) zostaje zjedzony(a) przez " << organizm->getNazwa() << " ( " << organizm->getPole().first + 1 << "," << organizm->getPole().second + 1 << " ) ";
		swiat->getSwiat()[pole.first][pole.second]->setZyje(false);
		swiat->getSwiat()[pole.first][pole.second] = new PustePole();
	}
	return "Przegrywa";
}

string WilczeJagody::getNazwa() {
	return "wilcze jagody";
}

File: Projekt2_java/src/projekt2/rosliny/WilczeJagody.java
package projekt2.rosliny;

import projekt2.*;

import java.awt.*;

public class WilczeJagody extends Roslina {
    public WilczeJagody(Pair pole, Swiat swiat) {
        super(99, 0, pole, swiat);
    }

    @Override
    public Organizm potomek(Pair pole) {
        return new WilczeJagody(pole, swiat);
    }

    @Override
    public Color rysuj() {
        return Color.BLUE.darker();
    }

    @Override
    public String rodzajKolizji(Organizm organizm) {
        super.rodzajKolizji(organizm);
        return "Ginie";
    }

    @Override
    public String getNazwa() {
        return "wilcze jagody";
    }
}
